apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js

```diff
@@ -36,43 +47,98 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
                 const { data } = args
                 const { address, unitName, unitType, billingAccount, resident, dv, sender } = data
 
-                const residents = await Resident.getAll(context,
-                    resident ? { id: resident.id, deletedAt: null } : {
-                        address: address,
-                        unitName: unitName,
-                        unitType: unitType,
-                        deletedAt: null,
-                    })
-
-                const billingAccounts = await BillingAccount.getAll(context,
-                    billingAccount ? { id: billingAccount.id, deletedAt: null } : {
-                        context: { status: 'Finished', deletedAt: null },
-                        property: { address: address, deletedAt: null  },
-                        unitName: unitName,
-                        unitType: unitType,
+                const reqId = get(context, ['req', 'id'])
+
+                const billingAccounts = await BillingAccount.getAll(
+                    context,
+                    {
                         deletedAt: null,
-                    })
+                        context: { status: CONTEXT_FINISHED_STATUS, deletedAt: null },
+                        ...billingAccount ? {
+                            id: billingAccount.id,
+                        } : {
+                            property: { address, deletedAt: null }, unitName, unitType,
+                        },
+                    },
+                )
 
+                const residents = await Resident.getAll(
+                    context,
+                    { deletedAt: null, ...resident ? { id: resident.id } : { address, unitName, unitType } },
+                )
 
-                const getPayload = (accountNumber, resident) => ({
-                    dv,
-                    sender,
-                    resident: { connect: { id: resident.id } },
-                    organization: { connect: { id: get(resident, ['organization', 'id']) } },
-                    accountNumber: accountNumber,
-                })
+                // Keep only residents of organizations the feature flag enabled for
+                const residentsFilteredByFeatureFlag = await asyncFilter(
```

Comment:
WTF?

Reply:
It's not possible to filter residents like this.

1. Registration for `SERVICE_PROVIDER`s will be open to anyone
2. Residents are only linked to `MANAGING_COMPANY` organisations, which means this filter will not even work